### PR TITLE
Task 44467 :Product permalink

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 1]
```diff
@@ -891,6 +891,17 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
     background-color: @disabledBackgroundButtonDefault !important;
     pointer-events: auto !important;
   }
+  #buyFormProduct {
+    .quantity {
+      max-width: 80%;
+    }
+    .priceSymbol {
+      color: #d07b7b;
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/css/main.less]: We need to use variabels

File: perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue [hunk 2]
```diff
@@ -377,9 +394,12 @@ export default {
           .replace(/=/g, '":"')}"}`
       );
     }
-    return this.init(parameters && parameters.productId, parameters && parameters.orderId, parameters && parameters.notProcessedOrders && parameters.notProcessedOrders === 'true');
+    return  this.init(parameters && parameters.productId, parameters && parameters.orderId, parameters && parameters.notProcessedOrders && parameters.notProcessedOrders === 'true');
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue]:
```suggestion
    return this.init(parameters && parameters.productId, parameters && parameters.orderId, parameters && parameters.notProcessedOrders && parameters.notProcessedOrders === 'true');
```

File: perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue [hunk 3]
```diff
@@ -395,7 +415,7 @@ export default {
       this.products = [];
       this.productsFilters = getProductFilter() || {};
       this.loading = true;
-      return initSettings()
+      return  initSettings()
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue]:
```suggestion
      return initSettings()
```

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue [hunk 4]
```diff
@@ -15,119 +15,123 @@ along with this program; if not, write to the Free Software Foundation,
...
-          <v-flex
-            v-if="productImage"
-            class="d-sm-flex d-none"
-            sm5>
+  <v-form id="buyFormProduct" ref="form">
+    <v-container>
+      <v-flex v-if="error && !loading" xs12>
+        <div class="alert alert-error v-content">
+          <i class="uiIconError"></i>
+          {{ error }}
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue:28]: We have already a container for errors, we can use it.

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue [hunk 5]
```diff
@@ -15,119 +15,123 @@ along with this program; if not, write to the Free Software Foundation,
...
+        <label class="pl-4 pb-2 font-weight-bold">{{ $t('exoplatform.perkstore.label.Gallery') }}</label>
+      </v-row>
+      <v-row v-if="productImage">
+        <v-col
+          v-for="img in product.imageFiles.slice(0, 3)"
+          :key="img.src"
+          class="pl-4"
+          cols="12"
+          sm="4">
+          <a :href="img.src" target="_blank">
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue:47]: Is the link required to open the image in a new tab ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue [hunk 6]
```diff
@@ -16,117 +16,115 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
-                  </v-list>
-                </v-menu>
-              </v-toolbar>
-            </v-carousel-item>
-          </template>
-        </v-carousel>
+    <v-card
+      :class="hideElevation && 'productDetailContentCard' || `elevation-${hover ? 9 : 1} productDetailContentCard`"
+      :max-width="cardHeight"
+      v-on="!canBuyProduct ? { click: displayBuyModal } : {}">
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue]: shouldn't be the opposite here ?
`v-on="canBuyProduct ? { click: displayBuyModal } : {}">`

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue [hunk 7]
```diff
@@ -192,6 +190,9 @@ export default {
     showMenu: false
   }),
   computed: {
+    canBuyProduct() {
```
Comment (ahamdi) [inline perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue]: either we change the name of the computed or we change the logic of the returned value

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue (hunk 5 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue:47]: yes

File: perk-store-webapps/src/main/webapp/css/main.less (hunk 1 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/css/main.less]: done !

File: perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue (hunk 2 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue]: done !

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue (hunk 4 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyForm.vue:28]: done

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue (hunk 6 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue]: changed to cantBuyproduct

File: perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue (hunk 7 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/vue-app/components/perk-store/ProductDetail.vue]: done

File: perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue (hunk 3 above)
Comment (Rdidmohamedelhadi) [reply perk-store-webapps/src/main/webapp/vue-app/components/PerkStoreApp.vue]: done

**Review by ahamdi — APPROVED** [review]

**Pull request merged**